app/controllers/api/likes_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Api::LikesController < Api::ApiController
+  before_action :authenticate_user!
+  before_action :set_post
+
+  def create
+    @like = @post.likes.where(user_id: current_user.id).first_or_create
+
+    unless @like.save
+      render json: { error: "Unable to create like" }, status: 422
+    end
+  end
+
+  def destroy
+    @like = @post.likes.where(user_id: current_user.id).destroy_all
```

Comment:
This should be: 

```ruby
@like = current_user.likes.find_by(post_id: @post.id)
@like.destroy

head 204
```

The first change because you always want to think about operating on the current users likes because if you don't, you could be operating on *any* like. You cover that case here but I feel like this format makes the behavior/intention more clear.

The `head 204` is nice because you're saying there's no content to send down